ANTLR/src/Compiler/SymbolTable/BaseScope.java

```diff
@@ -0,0 +1,47 @@
+package Compiler.SymbolTable;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class BaseScope implements Scope{
+    protected Scope enclosingScope;
+
+    protected Map<String, Symbol> symbols = new LinkedHashMap<>();
+
+    //Only use this to create the global scope
+    public BaseScope() { }
+
+    public BaseScope(Scope enclosingScope){
+        setEnclosingScope(enclosingScope);
+    }
+
+    @Override
+    public Scope getEnclosingScope() {
+        return enclosingScope;
```

Comment:
Don't we need a this keyword?
```suggestion
        return this.enclosingScope;
```